### DISCUSSION TITLE
Deep process pkt using nfstream using UDP

### DISCUSSION THREAD

**Discussion by stanpao** [nfstream/nfstream#85] [Q&A] [answered]
In such scenario:
nfstream process pcap file and extract them to flow data. During such process, nfstream uses C lang to process pkt content.I wonder if  I can define a UDP to use the raw pkt content extract pkt struct using other tools such as tshark ? Is it can be realized?

**Comment by aouinizied** [comment, marked as answer]
@stanpao Yes, you can use dpkt or scapy or whatever you want within your User Defined Plugin.
Here some example of UDPs we implemented: https://github.com/nfstream/nfstream/tree/master/nfstream/plugins

Zied
